### PR TITLE
sms_to_wordpress: upgrade IXR Library (using composer)

### PR DESCRIPTION
**Pull request by kingster**
Rewrite using composer #17

### CODE REVIEW

**Comment by tenzap** [comment]
I read somewhere that when using composer there is no need for all these require/include stuff. Maybe that was misunderstanding. Do you by chance know how to do this? I thought the change you suggested was about this.

Another thing that annoys me with composer is that it is not easy to package kalkun for the linux distributions. For example on debian, there is no internet access and the source package has to provide all the sources. Do you think we could still deliver versioned packages containing the libs taken with composer (codeigniter & this lib for now) ?

File: README.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ You need to install and configure this first:
    If you use a specific port with PostgreSQL, you may also need to set
    `$db['default']['port'] = "5432";`
 
-1. Import gammu database schema (it's included on gammu source, eg. `gammu/docs/sql/mysql.sql`)
+1. Import gammu database schema (it's included on gammu source, eg. `gammu/docs/sql/mysql.sql`) or download from `https://raw.githubusercontent.com/gammu/gammu/master/docs/sql/mysql.sql`
```
Comment (tenzap) [inline README.md]: I understand that it's useful to have such a link for reference, but I don't think it's a good idea to link the master branch of upstream. Nothing guaranties from a change in the schema and that the file is in line with the version of gammu-sms the user installed. It's better to redirect the user to the sources he used or the packaging system he uses.

File: application/plugins/sms_to_wordpress/sms_to_wordpress.php [hunk 2]
```diff
@@ -83,11 +84,12 @@ function sms_to_wordpress($sms)
 	{
 		$CI =& get_instance();
 		$CI->load->model('sms_to_wordpress/sms_to_wordpress_model', 'plugin_model');
-		require_once __DIR__.'/libraries/php-ixr-1.8.3/src/Client/Client.php';
-		require_once __DIR__.'/libraries/php-ixr-1.8.3/src/Message/Error.php';
-		require_once __DIR__.'/libraries/php-ixr-1.8.3/src/Message/Message.php';
-		require_once __DIR__.'/libraries/php-ixr-1.8.3/src/Request/Request.php';
-		require_once __DIR__.'/libraries/php-ixr-1.8.3/src/DataType/Value.php';
+
+		require_once 'vendor/kissifrot/php-ixr/src//Client/Client.php';
```
Comment (tenzap) [inline application/plugins/sms_to_wordpress/sms_to_wordpress.php]: maybe replace "//" by "/" ?

File: README.md (hunk 1 above)
Comment (kingster) [reply README.md]: Aggred! Removing this.

File: application/plugins/sms_to_wordpress/sms_to_wordpress.php (hunk 2 above)
Comment (kingster) [reply application/plugins/sms_to_wordpress/sms_to_wordpress.php]: Done, copy paste issue :(

**Comment by kingster** [comment]
> I read somewhere that when using composer there is no need for all these require/include stuff. Maybe that was misunderstanding. Do you by chance know how to do this? I thought the change you suggested was about this.

For autoloading, I think composer has that option, but that's something that we have to configure it. But even with basic vendoring it takes care to not having library code in the repo.

> Another thing that annoys me with composer is that it is not easy to package kalkun for the linux distributions. For example on debian, there is no internet access and the source package has to provide all the sources. Do you think we could still deliver versioned packages containing the libs taken with composer (codeigniter & this lib for now) ?

Not exactly. The CI step has a `dist` process. In that step it generates a full build with all output which you can copy to any air-gaped system and still run the full application.

File: .github/workflows/codeigniter.yml [hunk 3]
```diff
@@ -2,7 +2,11 @@
 name: Testing Kalkun
 on:
   push:
-    branches: [ "*" ]
+    branches: 
+      - master
+      - devel
+      - 'release/**'
```
Comment (tenzap) [inline .github/workflows/codeigniter.yml]: In some cases (don't remember exactly which one) the "/" is not liked by git. Could we use something else instead? underscore? dot? dash? whatever

**Comment by tenzap** [comment]
> Not exactly. The CI step has a `dist` process. In that step it generates a full build with all output which you can copy to any air-gaped system and still run the full application.

So is this the file that will be published as packaged source release?

File: .github/workflows/codeigniter.yml (hunk 3 above)
Comment (kingster) [reply .github/workflows/codeigniter.yml]: Sure

**Comment by kingster** [comment]
> So is this the file that will be published as packaged source release?

Yes.

File: composer.lock [hunk 4]
```diff
@@ -3764,4 +3764,4 @@
     "platform": [],
     "platform-dev": [],
     "plugin-api-version": "2.0.0"
-}
+}
```
Comment (tenzap) [inline composer.lock]: Its probably composer which does that, but as a convention could we agree that all files should end with a new line (this is checked for the other files during code style check)? The editors may also be in cause. On my systems, it always saves with empty new line.

Comment (kingster) [reply composer.lock]: Hmm, true. But we may not control all library/external files. :)

Comment (tenzap) [reply composer.lock]: :D

**Review by tenzap — APPROVED** [review]

**Comment by tenzap** [comment]
This one should be squashed (too many add/delete of files in each commit) ;)
I let you do it?

**Comment by tenzap** [comment]
~~I just noticed that your code doesn't work for me (this plugin is a bit tricky to test)
That's why I used the absolute path to the files of the lib.~~

```
 Severity: Warning --> require_once(vendor/kissifrot/php-ixr/src/Client/Client.php): failed to open stream: No such file or directory /---/Kalkun/application/plugins/sms_to_wordpress/sms_to_wordpress.php 88
ERROR - 2021-12-08 15:18:36 --> Severity: Compile Error --> require_once(): Failed opening required 'vendor/kissifrot/php-ixr/src/Client/Client.php' (include_path='.:/usr/share/php') /---/Kalkun/application/plugins/sms_to_wordpress/sms_to_wordpress.php 88
```

~~Either make it relative or absolute.~~

No, that's my error. I forgot to update composer :D Sorry

**Pull request merged**